test/plugin.test.js

```diff
@@ -62,6 +62,80 @@ test('fastify.register with fastify-plugin should not incapsulate his code', t =
   })
 })
 
+test('fastify.register with fastify-plugin should provide access to external fastify instance if opts argument is a function', t => {
+  t.plan(20)
+  const fastify = Fastify()
+
+  fastify.register((instance, opts, next) => {
+    instance.register(fp((i, o, n) => {
```

Comment:
can you use more descriptive variable names?

Reply:
On it, I will reply to the thread in total as some potentially unwanted or interesting behaviour arises from the `avvio` update.